### ISSUE TITLE
Should the 1 billion row file be deterministic?

### ISSUE THREAD

**Issue by datdenkikniet** [gunnarmorling/1brc#35]
Currently it seems that the 1 billion rows file is generated randomly. Making the generation pseudorandom would make sharing the 1 billion row file a little easier, and would make sure that everyone is running exactly the same test.

Just using a `Random` with a predefined seed to pick out numbers, and seeding a `Random` with the hash code of the city name to obtain measurements should do the trick.

**Comment by datdenkikniet** [comment]
I've now opened a PR that adds this functionality in #149. Also puts in a little bit of ground work to hopefully make #125 a bit easier to use generically.

**Comment by mtopolnik** [comment]
The evaluation shouldn't use a public test file because that allows the contenders to tightly optimize for the exact keyset in that file. For example, tweaking the hash function to minimize collisions, having special cases for some keys, sizing everything exactly right for the keyset, etc.

**Comment by mtopolnik** [comment]
If there's concern that some solution may just get unlucky with a given keyset, the winner can be determined by repeating the test with 2-3 different test files. I very much doubt that this would be a factor, given the large keyset size (10,000); more noise can be expected from all the environmental factors on the test machine.

**Comment by datdenkikniet** [comment]
I agree! I do not think that the current test-file should be shared or changed, but am asking for determinism so that it becomes a lot easier to compare/run on 1 billion row files that other contestants are using without requiring transmission of the entire data file.
